components/list/src/List.js

```diff
@@ -0,0 +1,47 @@
+/* @jsx jsx */
+
+import React, { Children, cloneElement } from 'react';
+import PropTypes from 'prop-types';
+import { jsx, css } from '@westpac/core';
+import { ListItem } from './ListItem';
+
+// ==============================
+// Component
+// ==============================
+export const List = ({ children, ...props }) => {
+	const childrenWithProps = Children.map(children, child =>
+		child.props.type === 'icon' && child.props.icon
+			? cloneElement(child, { ...props, icon: child.props.icon })
+			: cloneElement(child, props)
+	);
+
+	const { type } = props;
+	const ListType = type === 'ordered' ? 'ol' : 'ul';
+	return (
+		<ListType css={{ margin: 0, padding: type === 'ordered' ? '0 0 0 20px' : 0 }}>
+			{childrenWithProps}
+		</ListType>
+	);
+};
+
+// ==============================
+// Types
+// ==============================
+List.propTypes = {
+	/** The appearance of the bullet list */
+	appearance: PropTypes.oneOf(['primary', 'hero', 'neutral']),
+	/** The type of the bullet */
+	type: PropTypes.oneOf(['bullet', 'link', 'tick', 'unstyled', 'icon', 'ordered']),
+	/** The size of space between list elements */
+	spacing: PropTypes.oneOf(['medium', 'large']),
+	/** The icon for list */
+	icon: PropTypes.func,
+	/**  Any renderable child */
+	children: PropTypes.node,
+};
```

Comment:
I've been adding a note to say the prop is passed to children, so it's communicated in docz. Thoughts?

Reply:
It looked ugly repeating for each prop so I put a message above the props table in docs saying that the styling props will be passed to children

Reply:
I have a pattern in forms where only a subset of the props are passed down so made sense for that. Managing this message in the component vs in docs is I guess is the thing to consider.

Let's leave what you've got and add it to discuss